tests/test_consumer.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from vessim.core.consumer import MockPowerMeter, ComputingSystem
+
+class TestMockPowerMeter:
+
+    @pytest.fixture
+    def power_meter(self) -> MockPowerMeter:
+        return MockPowerMeter(p=20, power_config={
+            "high performance": 1,
+            "normal": .8,
+            "power-saving": .4
+        })
+
+    def test_initialize_fails_with_invalid_p_value(self):
+        with pytest.raises(ValueError):
+            MockPowerMeter(p=-1.0)
+
+    def test_initialize_fails_with_invalid_power_config(self):
```

Comment:
very good use case to use pytest's parametrize: https://docs.pytest.org/en/7.3.x/how-to/parametrize.html :)